### PR TITLE
Feat/update textreplacement

### PR DESCRIPTION
**Pull request by karlknowit**
## Explain the changes you’ve made
Update text replacement to include `#partnerName` and `#encryptionPin`. Also added the ability to do text replacement in the `components` property in a quetion.

## Explain why these changes are made
Since the `Card` component in form-builder needs to be able to show encryption pin and partner name, the textreplacement logic needed to be updated.

## Explain your solution
* Included the `components` property in textReplacement
* Added new rule to replacing encryption code
* Provide encryptionPin to the `Form` component to be able to do text replacement on that
* Update tests

## How to test

1. Checkout this branch
2. Fire upp the simulator by running the command `yarn ios` or `yarn android`
3. Make sure you have a form with the `Card` component
4. include `#encryptionPin` and/or `#partnerName` in the text in the component
5. Run the form in a simulator and make sure that the encryption pin is shown 

## Tested environments

- [] Storybook on a iOS device/simulator.
- [x] Building the Application on a iOS device/simulator.
- [] Building the Application on a Android device/simulator.

## Screenshots

image

### CODE REVIEW

**Comment by helsingborgsstad** [comment]
Task linked: [CU-23v1qc0 Visa encryption pin och partnername i form-builder card component](https://app.clickup.com/t/23v1qc0)

**Review by Sven65 — APPROVED** [review]

File: source/screens/FormCaseScreen.tsx [hunk 1]
```diff
@@ -54,30 +65,46 @@ const FormCaseScreen = ({
...
+            };
+          }
+
+          const encryption =
+            initCase?.forms?.[initCase.currentFormId].encryption ?? {};
 
-        void getForm(initCase.currentFormId);
-        setCurrentFormId(initCase.currentFormId);
+          const pinCode =
+            ((await getPasswordForForm(
```
Comment (MasterKenth) [inline source/screens/FormCaseScreen.tsx:94]: `getPasswordForForm` is designed to throw when an "invalid" form is provided. Solo form (form without co-applicant) is considered invalid so it fails here. 

We could use `to` to catch this:

```typescript
const [, password] = await to(getPasswordForForm(...))
const passwordTouse password ?? ""
```

**Review by MasterKenth — CHANGES_REQUESTED** [review]
Does not work with solo applicant due to an error. Other than that seems to work as expected.

**Comment by karlknowit** [comment]
Good enough!

**Comment by karlknowit** [comment]
image

**Pull request merged**